src/sql_engine/src/dml/update.rs

```diff
@@ -91,43 +62,37 @@ impl UpdateCommand {
         }
 
-        let all_columns = self.storage.table_columns(&schema_name, &table_name)?;
-        let mut errors = Vec::new();
-        let mut index_value_pairs = Vec::new();
+
+        let all_columns;
+        // let mut errors = Vec::new();
         let mut non_existing_columns = BTreeSet::new();
         let mut column_exists = false;
 
-        // only process the rows if the table and schema exist.
-        if self.storage.table_exists(&schema_name, &table_name) {
-            for (column_name, value) in to_update {
-                for (index, column_definition) in all_columns.iter().enumerate() {
-                    if column_definition.has_name(&column_name) {
-                        let v = match value.clone() {
-                            Value::Number(v) => v.to_string(),
-                            Value::SingleQuotedString(v) => v.to_string(),
-                            Value::Boolean(v) => v.to_string(),
-                            _ => unimplemented!("other types not implemented"),
-                        };
-                        match column_definition.sql_type().constraint().validate(v.as_str()) {
-                            Ok(()) => {
-                                index_value_pairs.push((index, Datum::try_from(&value).unwrap()));
-                            }
-                            Err(e) => {
-                                errors.push((e, column_definition.clone()));
-                            }
-                        }
-
-                        column_exists = true;
-
-                        break;
-                    }
-                }
 
-                if !column_exists {
-                    non_existing_columns.insert(column_name.clone());
-                }
-            }
-        } else {
+        // only process the rows if the table and schema exist.
+        if !self.storage.table_exists(&schema_name, &table_name) {
+            // for (column_name, value) in to_update {
+            //     for (index, column_definition) in all_columns.iter().enumerate() {
+            //         if column_definition.has_name(&column_name) {
+            //             let datum = value.as_datum().unwrap();
+            //             let v = datum.to_string();
+            //             match column_definition.sql_type().constraint().validate(v.as_str()) {
+            //                 Ok(()) => {
+            //                     index_value_pairs.push((index, datum));
+            //                 }
+            //                 Err(e) => {
+            //                     errors.push((e, column_definition.clone()));
+            //                 }
+            //             }
+            //             column_exists = true;
+            //             break;
+            //         }
+            //     }
+            //
+            //     if !column_exists {
+            //         non_existing_columns.insert(column_name.clone());
+            //     }
+            // }
```

Comment:
Uhhh 😞, I suppose it makes some constraint validation tests to fail. I believe this is the reason why I was against using `Datum` in expression evaluation because I will trigger big changes over the project. 
Either we 1) change `column_definition.sql_type().constraint().validate(v.as_str())` or 2) ignore tests. I am not in favour of either option. Because 1) make it hard to review/merge PRs, 2) if we do it once then we think this is normal to reduce already working functionality which I don't like.

Reply:
BTW, have you tried `Datum.to_string().as_str()`? will it help with tests? PR is growing touching more and more code base.
It is more comfortable with smaller changes, as every line of code could influence many things 🤷

Reply:
I guess I can leave the changes I have in just so you can see what I was thinking, but I will refactor it to use `sql_type.constraint().validate()` method for the time being. Validation using a string isn't performant nor does it make sense.

Reply:
> Validation using a string isn't performant nor does it make sense.

I triggered the thought that there is no software application that "complete" 😄 . I, even, am not sure that there is anything done by a human is "complete", I mean constructed as what a creator would like to see creature 😄